assembl/static2/js/app/components/common/socialShare.jsx

```diff
@@ -0,0 +1,96 @@
+const FacebookIcon = generateShareIcon('facebook');
+const GooglePlusIcon = generateShareIcon('google');
+const LinkedinIcon = generateShareIcon('linkedin');
+const TwitterIcon = generateShareIcon('twitter');
+const EmailIcon = generateShareIcon('email');
+const WhatsappIcon = generateShareIcon('whatsapp');
+const TelegramIcon = generateShareIcon('telegram');
+
+export default class SocialShare extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      copied: false
+    };
+  }
+  render() {
+    const { url, onClose, social } = this.props;
+    return (
+      <div className="share-buttons-container">
+        {social
+          ? <div className="social-share-buttons-container">
+            <div className="social-share-button">
+              <EmailShareButton url={url} onShareWindowClose={onClose}>
+                <EmailIcon size={32} round />
+              </EmailShareButton>
+            </div>
+            <div className="social-share-button">
+              <FacebookShareButton url={url} onShareWindowClose={onClose}>
+                <FacebookIcon size={32} round />
+              </FacebookShareButton>
+            </div>
+            <div className="social-share-button">
+              <GooglePlusShareButton url={url} onShareWindowClose={onClose}>
+                <GooglePlusIcon size={32} round />
+              </GooglePlusShareButton>
+            </div>
+
+            <div className="social-share-button">
+              <LinkedinShareButton url={url} onShareWindowClose={onClose}>
+                <LinkedinIcon size={32} round />
+              </LinkedinShareButton>
+            </div>
+
+            <div className="social-share-button">
+              <TwitterShareButton url={url} onShareWindowClose={onClose}>
+                <TwitterIcon size={32} round />
+              </TwitterShareButton>
+            </div>
+
+            <div className="social-share-button">
+              <WhatsappShareButton url={url} onShareWindowClose={onClose}>
+                <WhatsappIcon size={32} round />
+              </WhatsappShareButton>
+            </div>
+            <div className="social-share-button">
+              <TelegramShareButton url={url} onShareWindowClose={onClose}>
+                <TelegramIcon size={32} round />
+              </TelegramShareButton>
+            </div>
+          </div>
+          : <div className="social-share-buttons-container">
+            <div className="social-share-button">
+              <EmailShareButton url={url} onShareWindowClose={onClose}>
+                <EmailIcon size={32} round />
+              </EmailShareButton>
+            </div>
+          </div>}
+        <CopyToClipboard
+          text={url}
+          onCopy={() => {
+            return this.setState({ copied: true });
+          }}
+        >
+          <button className="btn btn-default btn-copy">
+            {this.state.copied ? 'Link copied' : 'Copy link to Clipboard'}
```

Comment:
please translate this with Ii18n.t

Reply:
I don't think it make sense to change the label of the button here. You can add a message 'Link copied' next to the button if you want, but I think the button should keep "Copy link to Clipboard".

Reply:
I had done this at first (the message showing after the click) the thing is the modal is automatically growing when you add text inside it and it made a weird effect. So Roxana wants it this way as an MVP and she's creating a ticket with the requirements for something more sophisticated (icons in the button and a tooltip for "link copied" message)

Reply:
write a '&nbsp;' maybe so that the place is reserved so the modal won't change size?
Having a button with 'Link copied' that when you click copy the link is a no no for me.

Reply:
You could use a green alert to signal a copy (like giphy), or do as @vincentfretin suggested. Vincent, for the sake of Milan's learning, we should clarify why something is a no no.

Reply:
Seriously? This is common sense. It's a button with "link copied" label! what does it means? What does it do when I click on it? A button needs to describe an action, should be a verb.

Reply:
I don't disagree with you, it's just that Roxana liked it this way so I have left like that in the meantime (until we get her favourite solution which she made a mock up for). I can do what you said but it can be confusing to know what to do when getting different requests...

Reply:
Sure, I understand. Let's keep it that way for now.

Reply:
Guys, at the end, it's the PO who decides even if you disagree. Of course you can give your opinion and the most of time the PO will follow it.